src/sdam/description/topology/mod.rs

```diff
@@ -20,63 +20,87 @@ use crate::{
 
 const DEFAULT_HEARTBEAT_FREQUENCY: Duration = Duration::from_secs(10);
 
-/// The TopologyType type, as described by the SDAM spec.
+/// The possible types for a topology.
 #[derive(Debug, Clone, Copy, Eq, PartialEq, Deserialize)]
-pub(crate) enum TopologyType {
+#[non_exhaustive]
+pub enum TopologyType {
+    /// A single mongod server.
     Single,
+
+    /// A replica set with no primary.
     ReplicaSetNoPrimary,
+
+    /// A replica set with a primary.
     ReplicaSetWithPrimary,
+
+    /// A sharded topology.
     Sharded,
+
+    /// A topology whose type is not known.
     Unknown,
 }
 
+#[cfg(test)]
+impl TopologyType {
+    fn as_str(&self) -> &str {
```

Comment:
I think this will infer the output string to have the same lifetime as `self`; it'll be more useful as `&'static str`.